pkg/operator/ceph/cluster/mgr/dashboard.go

```diff
@@ -201,26 +203,54 @@ func (c *Cluster) setLoginCredentials(password string) error {
 	// Set the login credentials. Write the command/args to the debug log so we don't write the password by default to the log.
 	logger.Infof("setting ceph dashboard %q login creds", dashboardUsername)
 
-	// retry a few times in the case that the mgr module is not ready to accept commands
-	_, err := client.ExecuteCephCommandWithRetry(func() (string, []byte, error) {
-		args := []string{"dashboard", "set-login-credentials", dashboardUsername, password}
-		cmd := client.NewCephCommand(c.context, c.clusterInfo, args)
-		output, err := cmd.RunWithTimeout(client.CmdExecuteTimeout)
-		return "set dashboard creds", output, err
-	}, c.exitCode, 5, invalidArgErrorCode, dashboardInitWaitTime)
-	if err != nil {
-		return errors.Wrapf(err, "failed to set login creds on mgr")
+	cephVersion := c.spec.CephVersion.Image
+	if cephVersion == "ceph/ceph:v15.2.10" || cephVersion == "ceph/ceph:v14.2.17" {
```

Comment:
To check for a ceph version that is at least these versions, I'd suggest a helper bool method. Also see examples where we call something like: `version.IsAtLeast(cephver.CephVersion{Major: 14, Minor: 2, Extra: 13})`. This checks the actual version instead of just the image tag.

Reply:
right, I was also looking for a better way. Thanks for the suggestion.
Added